pyfstat/make_sfts.py

```diff
@@ -272,157 +293,29 @@ def basic_setup(self):
     def tend(self):
         return self.tstart + self.duration
 
-    def make_data(self, verbose=False):
-        """ A convienience wrapper to generate a cff file then sfts """
-        if self.h0:
-            self.make_cff(verbose)
-        else:
-            logging.info("Got h0=0, not writing an injection .cff file.")
-        self.run_makefakedata()
-
-    def get_base_template(self, i, Alpha, Delta, h0, cosi, psi, phi, F0, F1, F2, tref):
-        return """[TS{}]
-Alpha = {:1.18e}
-Delta = {:1.18e}
-h0 = {:1.18e}
-cosi = {:1.18e}
-psi = {:1.18e}
-phi0 = {:1.18e}
-Freq = {:1.18e}
-f1dot = {:1.18e}
-f2dot = {:1.18e}
-refTime = {:10.6f}"""
-
-    def get_single_config_line_cw(
-        self, i, Alpha, Delta, h0, cosi, psi, phi, F0, F1, F2, tref
-    ):
-        template = (
-            self.get_base_template(
-                i, Alpha, Delta, h0, cosi, psi, phi, F0, F1, F2, tref
-            )
-            + """\n"""
-        )
-        return template.format(i, Alpha, Delta, h0, cosi, psi, phi, F0, F1, F2, tref)
-
-    def get_single_config_line_tcw(
-        self,
-        i,
-        Alpha,
-        Delta,
-        h0,
-        cosi,
-        psi,
-        phi,
-        F0,
-        F1,
-        F2,
-        tref,
-        window,
-        transientStartTime,
-        transientTau,
-    ):
-        template = (
-            self.get_base_template(
-                i, Alpha, Delta, h0, cosi, psi, phi, F0, F1, F2, tref
-            )
-            + """
-transientWindowType = {:s}
-transientStartTime = {:10.0f}
-transientTau = {:10.0f}\n"""
-        )
-        return template.format(
-            i,
-            Alpha,
-            Delta,
-            h0,
-            cosi,
-            psi,
-            phi,
-            F0,
-            F1,
-            F2,
-            tref,
-            window,
-            transientStartTime,
-            transientTau,
-        )
-
-    def get_single_config_line(
-        self,
-        i,
-        Alpha,
-        Delta,
-        h0,
-        cosi,
-        psi,
-        phi,
-        F0,
-        F1,
-        F2,
-        tref,
-        window,
-        transientStartTime,
-        transientTau,
-    ):
-        if window == "none":
-            return self.get_single_config_line_cw(
-                i, Alpha, Delta, h0, cosi, psi, phi, F0, F1, F2, tref
-            )
-        else:
-            return self.get_single_config_line_tcw(
-                i,
-                Alpha,
-                Delta,
-                h0,
-                cosi,
-                psi,
-                phi,
-                F0,
-                F1,
-                F2,
-                tref,
-                window,
-                transientStartTime,
-                transientTau,
-            )
-
-    def make_cff(self, verbose=False):
-        """
-        Generates a .cff file
-
+    def parse_args_consistent_with_mfd(self):
```

Comment:
Here's where all the magic happens: This method looks into the class itself using the `__dict__` attribute. If any of the attributes specified in `signal_parameter_labels` are not `None`, they are included into any output of
`get_single_config_line`, which is the main way for MFD to get signal parameters.

This way, `BinaryModulatedWriter` only needs to include its own parameters and nothing more.

Reply:
Makes a lot of sense!